### PR TITLE
Feature/CVSB-10362 - view a specialist test record

### PR DESCRIPTION
**Pull request by irina005**
Ticket:
https://jira.dvsacloud.uk/browse/CVSB-10362

Bugs solved in here:

- https://jira.dvsacloud.uk/browse/CVSB-17104

### CODE REVIEW

File: src/app/test-record/defects/defects.component.spec.ts [hunk 1]
```diff
@@ -36,4 +39,14 @@ describe('DefectsComponent', () => {
 
     expect(fixture).toMatchSnapshot();
   });
+
+  it('should match snapshot if test type is among specialist tests', () => {
```
Comment (alleypa) [inline src/app/test-record/defects/defects.component.spec.ts:43]: `should generate test type for specialist tests`. the snapshot file is your outcome

File: src/app/test-record/test-record.mapper.ts [hunk 2]
```diff
@@ -3,14 +3,17 @@ import { TestResultModel } from '@app/models/test-result.model';
 import { TEST_STATION_TYPE } from '@app/test-record/test-record.enums';
```
Comment (alleypa) [inline src/app/test-record/test-record.mapper.ts:3]: file need discussion

File: src/app/utils/testing-test-models.utils.ts [hunk 3]
```diff
@@ -27,3 +29,15 @@ function mockTestType(args?: Partial<TestType>): TestType {
...
+    testSectionApplicable1: { 43: 'test' },
+    testSectionApplicable2: { 43: 'test' },
+    specialistCertificateNotApplicable: { 155: 'test' },
+    seatBeltApplicable: { 43: 'test' },
+    defectsApplicable: { 43: 'test' },
+    emissionDetailsApplicable: { 44: 'test' },
+    specialistTestApplicable: { 125: 'test' },
+    specialistCOIFApplicable: { 176: 'test' }
+  } as TestTypesApplicable;
+}
```
Comment (alleypa) [inline src/app/utils/testing-test-models.utils.ts:43]: this is not a test-model

File: src/app/test-record/test-section/test-section.component.html [hunk 4]
```diff
@@ -39,19 +39,29 @@
...
+        {{ testType.certificateNumber | DefaultNullOrEmpty }}
+      </dd>
+    </div>
+  </ng-container>
   <ng-container *ngIf="testTypesApplicable">
     <div
       *ngIf="
-        testTypesApplicable.anniversaryDateApplicable[testType.testTypeId] ||
-        testTypesApplicable.certificateExtraApplicable[testType.testTypeId]
+        testTypesApplicable.specialistCOIFApplicable[testType.testTypeId] &&
```
Comment (JonLazariniDD) [inline src/app/test-record/test-section/test-section.component.html:55]: What about moving those into a service now?
They are very likely to be reused across and we could easily be testing them.
We could also be adding the future ones we will be having depending on the groups.

File: src/app/test-record/test-section/test-section.component.ts [hunk 5]
```diff
@@ -13,7 +13,17 @@ export class TestSectionComponent implements OnInit {
   @Input() testRecord: TestResultModel;
   @Input() testTypesApplicable: TestTypesApplicable;
 
+  hasCertificateNumber: boolean;
+
   constructor() {}
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.hasCertificateNumber =
```
Comment (JonLazariniDD) [inline src/app/test-record/test-section/test-section.component.ts:21]: Could this be moved into a service perhaps?
We could be using composition (with lodash?) later and probably chain the this.testTypesApplicable. assuming they return the same data type.

File: src/app/test-record/test-section/test-section.component.html (hunk 4 above)
Comment (JonLazariniDD) [reply src/app/test-record/test-section/test-section.component.html:55]: reference ticket: https://jira.dvsacloud.uk/browse/CVSB-16312

File: src/app/test-record/test-section/test-section.component.ts (hunk 5 above)
Comment (JonLazariniDD) [reply src/app/test-record/test-section/test-section.component.ts:21]: reference ticket: https://jira.dvsacloud.uk/browse/CVSB-16314

**Comment by JonLazariniDD** [comment]
let's please a meaningful PR description of what the code does as well as the automated test suite report and passing. We can check with testers if required if we need assistance.

**Pull request merged**